website/project/model.py

```diff
@@ -3089,13 +3131,20 @@ def to_json(self):
 
 
 class Sanction(StoredObject):
-    """Sanction object is a generic way to track approval states"""
-
+    """Sanction class is a generic way to track approval states"""
+    # Tell modularodm not to attach backends
     abstract = True
 
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+
+    # Neither approved not cancelled
     UNAPPROVED = 'unapproved'
+    # Has approval
     APPROVED = 'approved'
+    # Rejected by at least one person
     REJECTED = 'rejected'
+    # One of 'unapproved', 'active', 'cancelled', or 'completed
+    state = fields.StringField(default=UNAPPROVED)
```

Comment:
The recently merged `choice_in` validator makes it easier to add validation here. A comment line should not be the final word in data validation.